policyengine_core/simulations/simulation.py

```diff
@@ -1395,77 +1412,28 @@ def extract_person(
 
         return json.loads(json.dumps(situation, cls=NpEncoder))
 
-    def _get_macro_cache(
-        self,
-        variable_name: str,
-        period: str,
-    ):
+    def check_macro_cache(self, variable_name: str, period: str) -> bool:
         """
-        Get the cache location of a variable for a given period, if it exists.
+        Check if the variable is able to have cached value
         """
-        if not self.is_over_dataset:
-            return None
+        is_cache_available = True
+        if self.is_over_dataset:
+            return is_cache_available
 
         variable = self.tax_benefit_system.get_variable(variable_name)
         parameter_deps = variable.exhaustive_parameter_dependencies
 
         if parameter_deps is None:
-            return None
+            return not is_cache_available
```

Comment:
I think I would get rid of `is_cache_available` and just return either `True` or `False` in this function, as I think the `is_cache_available` value serves to complicate the function.